walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -27,124 +27,125 @@
-                TimeRange previousBlock = blockedTimesArrayList.get(blockIndex-1);
-                if (previousBlock.end() == currentBlock.start()) {
-                    continue;
-                }
-                start = previousBlock.end();
-                end = currentBlock.start();
-                toAdd = TimeRange.fromStartEnd(start, end, false);
-            }
-
-            // Make sure the non-blocked period is long enough for the meeting request
-            if (toAdd.duration() >= meetingDuration) {
-                timesToAdd.add(toAdd);
-            }
+      // Go through all the blocked times, add non-blocked periods to timesToAdd
+      for (int blockIndex = 0; blockIndex <= blockedTimesArrayList.size(); blockIndex++) {
+        int start;
+        int end;
+        TimeRange toAdd;
+
+        // If this is the first period of the day, start at 0
+        if (blockIndex == 0) {
+          TimeRange currentBlock = blockedTimesArrayList.get(blockIndex);
+          // If the first block starts at 0, skip
+          if (currentBlock.start() != TimeRange.START_OF_DAY) {
+            start = TimeRange.START_OF_DAY;
+            end = currentBlock.start();
+          } else {
+            continue;
+          }
+        } else if (blockIndex == blockedTimesArrayList.size()) {
+          // Check if the last block of the day ends before midnight, if it does, then add 
+          // a free period till midnight
+          TimeRange previousBlock = blockedTimesArrayList.get(blockIndex - 1);
+          if (previousBlock.end() != TimeRange.END_OF_DAY + 1) {
+            start = previousBlock.end();
+            end = TimeRange.END_OF_DAY;
+          } else {
+            continue;
+          }
+        } else {
+          // Create a new free period starting at the end of the previous event
+          // and ending at the start of the current event
+          TimeRange currentBlock = blockedTimesArrayList.get(blockIndex);
+          TimeRange previousBlock = blockedTimesArrayList.get(blockIndex - 1);
+          if (previousBlock.end() == currentBlock.start()) {
+            continue;
+          }
+          start = previousBlock.end();
+          end = currentBlock.start();
         }
+
+        toAdd = TimeRange.fromStartEnd(start, end, (blockIndex == blockedTimesArrayList.size()));
+
+        // Make sure the non-blocked period is long enough for the meeting request
+        if (toAdd.duration()>= meetingDuration) {
+          timesToAdd.add(toAdd);
+        }
+      }
     }
-    
+
     return timesToAdd;
   }
 
-  /* Get a sorted collection of all blocked ranges, removing nested and overlapping blocks */
-  private Collection<TimeRange> getBlockedTimeline(Collection<TimeRange> blockedTimes) {
+  /* Takes in a set of timeranges and parses them. Return a continuous, disjoint timeline of the events */
+  private Collection<TimeRange> process(Collection<TimeRange> blockedTimes) {
```

Comment:
maybe the name `process` doesn't explain what the function did. Maybe something like `MergeBlockTimes` makes more sense?

Reply:
Made the change!